src/bin/cql-stress-cassandra-stress/java_generate/distribution/normal.rs

```diff
@@ -56,3 +57,99 @@ impl Distribution for NormalDistribution {
         self.rng.get().set_seed(seed as u64)
     }
 }
+
+pub struct NormalDistributionFactory {
+    min: i64,
+    max: i64,
+    mean: f64,
+    standard_deviation: f64,
+}
+
+impl NormalDistributionFactory {
+    fn new(min: i64, max: i64, mean: f64, standard_deviation: f64) -> Result<Self> {
+        NormalDistribution::verify_args(min, max, standard_deviation)?;
+        Ok(Self {
+            min,
+            max,
+            mean,
+            standard_deviation,
+        })
+    }
+}
+
+impl DistributionFactory for NormalDistributionFactory {
+    fn create(&self) -> Box<dyn Distribution> {
+        Box::new(
+            NormalDistribution::new(self.min, self.max, self.mean, self.standard_deviation)
+                .unwrap(),
+        )
+    }
+}
+
+impl NormalDistributionFactory {
+    fn do_parse_from_description(desc: &Description<'_>) -> Result<Box<dyn DistributionFactory>> {
+        // See https://github.com/scylladb/scylla-tools-java/blob/master/tools/stress/src/org/apache/cassandra/stress/settings/OptionDistribution.java#L202.
+        let mut iter = desc.args.iter();
+
+        let (min, max) = match (iter.next(), iter.next()) {
+            (Some(min), Some(max)) => (min.parse::<i64>()?, max.parse::<i64>()?),
+            _ => anyhow::bail!("Expected at least 2 arguments"),
+        };
+
+        let (mean, stdev) = match (iter.next(), iter.next(), iter.next()) {
+            (Some(mean), Some(stdev), None) => (mean.parse::<f64>()?, stdev.parse::<f64>()?),
+            (maybe_stdvrng, None, None) => {
```

Comment:
In general, if one invocation of `iter.next()` returns `None`, the next one is _not guaranteed_ to return `None` again. See: https://doc.rust-lang.org/std/iter/trait.Iterator.html#tymethod.next

That said, I think it _should_ be the case for `desc.args.iter()` as it iterates over a Vec (slice), but it's not immediately obvious which kind iterator is that and, for the sake of clarity, I'd [fuse](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.fuse) it.